### PR TITLE
Idex 3528 Find Usages functionality does not work

### PR DESCRIPTION
**Pull request by evidolob**
Use websocket for search request.
Improve error handling and notification.
Fix search in multimodule projects.

### CODE REVIEW

File: plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModelManager.java [hunk 1]
```diff
@@ -2865,9 +2865,9 @@ public String toString() {
     }
 
     public JavaWorkspaceScope getWorkspaceScope() {
-        if (this.workspaceScope == null) {
+//        if (this.workspaceScope == null) {
```
Comment (skabashnyuk) [inline plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModelManager.java]: Can we remove commented code?

File: plugin-java/che-plugin-java-ext-lang-client/src/main/resources/org/eclipse/che/ide/ext/java/client/svg/find_usages.svg [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (skabashnyuk) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/resources/org/eclipse/che/ide/ext/java/client/svg/find_usages.svg]: who made this image?

File: plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModel.java [hunk 3]
```diff
@@ -203,10 +202,23 @@ public IJavaProject getJavaProject(IResource resource) {
...
+	int index = 0;
+	for (int i = 0; i < length; i++) {
+		IProject project = projects[i];
+		if (org.eclipse.jdt.internal.core.JavaProject.hasJavaNature(project)) {
+			children[index++] = getJavaProject(project);
+		}
+	}
+	if (index < length)
+		System.arraycopy(children, 0, children = new IJavaProject[index], 0, index);
+//    IJavaProject[] array = new IJavaProject[list.size()];
```
Comment (skabashnyuk) [inline plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModel.java]: can we   remove commented code?

File: plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModelManager.java (hunk 1 above)
Comment (evidolob) [inline plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModelManager.java]: no, it's eclipse code, https://jira.codenvycorp.com/browse/IDEX-3609 after this I return commented code

File: plugin-java/che-plugin-java-ext-lang-client/src/main/resources/org/eclipse/che/ide/ext/java/client/svg/find_usages.svg (hunk 2 above)
Comment (evidolob) [inline plugin-java/che-plugin-java-ext-lang-client/src/main/resources/org/eclipse/che/ide/ext/java/client/svg/find_usages.svg]: Stévan Le Meur give me this image

File: plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModel.java (hunk 3 above)
Comment (evidolob) [inline plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModel.java]: no, it's eclipse code, https://jira.codenvycorp.com/browse/IDEX-3609 after this I return commented code

**Comment by vparfonov** [comment]
Ok for me

**Pull request merged**